### ISSUE TITLE
[Feature request] add message option for the loading component

### ISSUE THREAD

**Issue by laurenmrice** [carbon-design-system/carbon#19882]
## Acceptance criteria

> [!Important]
>  - [x] Before implementing, this needs to be run by accessibility in a CAG or Design crit call. ([Figma file](https://www.figma.com/design/yGaC6LvEnUGEwAfQC5CEyg/Loading-message-options?node-id=0-1&t=4J7HNx7ER6pW658P-1))
>  - [ ] This issue ([Explore overlay token for better g100 contrast](https://github.com/carbon-design-system/carbon/issues/19898)) also will resolve the `$overlay` color token a11y contrast problem in the light themes.

----

- We have gotten a few requests and questions about whether we can add a message underneath the Loading indicator. People currently have to customize this one on their own.
- We should add an option to the storybook to have a message below the loading indicator.
- This is a common part of loading that I have seen across other systems, and it could be a helpful optional enhancement that our users could benefit from when using the loading component.
- Ref: [Figma product examples](https://www.figma.com/design/pGQqrdk9XcIqsybjiDf6Ao/Loading-docs?node-id=354-146835&t=X8oo4HLOTxTeckLc-0)

----

### Design spec
- [ ] Add any findings and suggestions here for this spec that came out of the $overlay token work. (Like what type token should be used, color token for the type and loading indicator across themes, etc.)
- [ ] Create final design spec

**Comment by sunny-babbar** [comment]
**Findings from the overlay color token project:**
--------

**Key links:**

- [[Overlay token] Audit #19996](https://github.com/carbon-design-system/carbon/issues/19996)
- [[Overlay token] Design iterations #19999](https://github.com/carbon-design-system/carbon/issues/19999)
- [[Overlay token] Create design spec #19997](https://github.com/carbon-design-system/carbon/issues/19997)

**Figma file link:**

- [Figma file link of all exploration](https://www.figma.com/design/1MWnprzFPwYpRfqNhUFpLq/-Overlay-token--Audit--19996?node-id=116-27390&t=TjL131ECSl1bjrRB-1)

--------------
**Proposed overlay token:**

- Black # 00000, @ 60% for all themes (white, gray 10, gray 90, gray 100)

<img width="24048" height="21264" alt="Image" src="https://github.com/user-attachments/assets/cad30f94-36c5-4ce1-9e77-56c8f46fd854" />

------------------
**Notes on loading overview of findings:** (Notes from Michael around a11y)

**Summary - Loading:**
- **Type token $heading-03:** Use larger text $heading-03 (If you are a bit under minimum contrast, larger text is better)
- **Type token:** There is an allowed reduction 3:1 contrast if the text is 14pt bold or 18pt - (Possibly try $heading-compact-01?)
- **Blue loading indicator meet 3:1:** If you always use text in combination with it, the loading symbol is decorative and does not need to meet any contrast.
Make sure the loading indicator is 3:1 always, whether there is text or not (this can be further investigated)
- **Color token $interactive:** Assume someone will sooner or later use the loading indicator without text, so it would be a good pro-active step to make sure the symbol does contrast 3:1 against the background. (Note: that if you do that, then at least for meeting the WCAG requirements, you could drop the visible text entirely (as long as there is and ALT for the loading image).
- **Loading state ends - new feature consideration:** Additional thought, there is no non-visual indicator of when the loading state ends. If you're sighted, you know it goes away and you can begin working much of the time, something will take focus and be announced by the screen reader at this point. But sometimes loading state being completed does not cause something to take focus. (Suggestion: A status message of "Loading completed" be generated. This does not have to be visible (although if visible text changed to that momentarily before the spinning wheel disappears, that might be a useful value add for everyone). A consideration as a new feature. **(Comments from Michael for a11y)** 

--------------
**Loading with text exploration**

- $body-compact-01
- $heading-compact-01
- $heading-03

**Note:**
- **Type token $heading-03:** Use larger text $heading-03 (If you are a bit under minimum contrast, larger text is better)
- **Type token:** There is an allowed reduction 3:1 contrast if the text is 14pt bold or 18pt - (Possibly try $heading-compact-01?)

<img width="24345" height="17283" alt="Image" src="https://github.com/user-attachments/assets/e52837db-5d6e-4bd1-86d3-9cefd27d60e3" />

------------

**Loading state ends - new feature consideration**

**Note:** 
- **Loading state ends - new feature consideration:** Additional thought, there is no non-visual indicator of when the loading state ends. If you're sighted, you know it goes away and you can begin working much of the time, something will take focus and be announced by the screen reader at this point. But sometimes loading state being completed does not cause something to take focus. (Suggestion: A status message of "Loading completed" be generated. This does not have to be visible (although if visible text changed to that momentarily before the spinning wheel disappears, that might be a useful value add for everyone). A consideration as a new feature. **(Comments from Michael for a11y)** 

<img width="24780" height="13665" alt="Image" src="https://github.com/user-attachments/assets/7950e1b5-a559-456b-9bd2-b05bcc42d428" />



FYI @laurenmrice
